lang/java/reef-webserver/src/main/java/org/apache/reef/webserver/HttpServerReefEventHandler.java

```diff
@@ -169,17 +170,17 @@ public void onHttpRequest(
       final List names = parsedHttpRequest.getQueryMap().get("filename");
       if (names == null || names.size() == 0) {
         response.getWriter().println(String.format("File name is not provided"));
-      }
-
-      final String fileName = (String)names.get(0);
-      if (!fileName.equals(driverStdoutFile) && !fileName.equals(driverStderrFile)) {
-        response.getWriter().println(String.format("Unsupported file names: [%s] ", fileName));
-      }
-      try {
-        final byte[] outputBody = readFile((String) names.get(0)).getBytes(StandardCharsets.UTF_8);
-        response.getOutputStream().write(outputBody);
-      } catch(final IOException e) {
-        response.getWriter().println(String.format("Cannot find the log file: [%s].", fileName));
+      } else {
+        final String fileName = (String) names.get(0);
+        if (!fileName.equals(driverStdoutFile) && !fileName.equals(driverStderrFile)) {
+          response.getWriter().println(String.format("Unsupported file names: [%s] ", fileName));
+        }
+        try {
+          final byte[] outputBody = readFile((String) names.get(0)).getBytes(StandardCharsets.UTF_8);
+          response.getOutputStream().write(outputBody);
+        } catch (final IOException e) {
+          response.getWriter().println(String.format("Cannot find the log file: [%s].", fileName));
```

Comment:
This logic seems a bit strange... See documentation for [`getOutputStream()`](https://tomcat.apache.org/tomcat-7.0-doc/servletapi/javax/servlet/ServletResponse.html#getOutputStream()): "Either this method or `getWriter()` may be called to write the body, not both."